source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -458,7 +458,8 @@ FilterTrailersStatus Filter::onTrailers(ProcessorState& state, Http::HeaderMap&
     return FilterTrailersStatus::StopIteration;
   }
 
-  if (!body_delivered && state.bufferedData() && state.bodyMode() == ProcessingMode::BUFFERED) {
+  if (!body_delivered && ((state.bufferedData() && state.bodyMode() == ProcessingMode::BUFFERED) ||
+                          (!state.chunkQueue().empty() && state.bodyMode() == ProcessingMode::BUFFERED_PARTIAL))) {
```

Comment:
BUFFERED_PARTIAL mode is missing here. Adding it back.

Reply:
For BUFFERED_PARTIAL mode, the data is stored in the chunk queue once it arrives. If the data never reaches queue limit, also no end_stream flag when receiving body,  the data needs to be delivered to the ext_proc server when trailer is received.

Logically, this is similar to the BUFFERED mode.  In BUFFERED mode, the data is buffered when it arrives. If no end_stream flag when receiving body, the data is delivered to the ext_proc server when trailer is received.  Such code is in place.